src/main/java/hibernate/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package hibernate.dao.impl;
+
+import hibernate.dao.MovieDao;
+import hibernate.lib.Dao;
+import hibernate.model.Movie;
+import hibernate.util.HibernateUtil;
+import java.util.List;
+import javax.persistence.criteria.CriteriaQuery;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert Movie entity", e);
```

Comment:
```suggestion
            throw new RuntimeException("Can't insert Movie entity", e);
```
throw your custom exception